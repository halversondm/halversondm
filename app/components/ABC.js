/**
 * Created by Daniel on 6/26/2016.
 */
"use strict";

import React from "react";

const ABC = React.createClass({
  render() {
    return <div>
    </div>;
  }
});

export default ABC;
