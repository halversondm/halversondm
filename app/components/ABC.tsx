/**
 * Created by Daniel on 6/26/2016.
 */
import * as React from 'react'
import { Modal } from 'react-bootstrap'
import { CheckboxSeries } from './CheckboxSeries'
import { type ReactNode } from 'react'

export interface ABCState {
  antecedentOtherDisabled: boolean
  messages: string[]
  showModal: boolean
  user: UserState
}

export interface UserState {
  when: string
  antecedent: string
  antecedentOther: string
  location: string
  people: string[]
  peopleOther: string
  behavior: string[]
  behaviorOther: string
  duration: string
  intensity: string
  consequence: string[]
  consequenceOther: string
}

export class ABC extends React.Component<unknown, ABCState> {
  public state: ABCState
  private people: CheckboxSeries
  private behaviors: CheckboxSeries
  private consequences: CheckboxSeries

  private readonly antecedentValues: string[] = ['Given Direction/task, asked to do something',
    'Asked to wait', 'Difficulty with task/activity',
    'Preferred activity interrupted', 'Activity/Item denied ("told no")',
    'Loud, noisy environment', 'Given assistance / correction',
    'Transition between locations', 'Attention given to others',
    'Attention not given when wanted', 'Left alone (no indiv. attention)']

  private readonly locationValues: string[] = ['Home', 'School', 'Other']
  private readonly peopleValues: string[] = ['Mom', 'Dad', 'Sibling', 'Grandparents', 'Alone', 'Peers']
  private readonly behaviorValues: string[] = ['Refuse to follow directions', 'Makes verbal threats',
    'Grabbing/pulling', 'Crying/Whining', 'Screaming/Yelling', 'Scratching',
    'Biting', 'Spitting', 'Kicking', 'Flopping', 'Running Away',
    'Destroying property', 'Hitting Self', 'Hitting Others',
    'Verbal Refusal']

  private readonly durationValues: string[] = ['< 1 min', '1 - 5 min', '5 - 10 min', '10 - 30 min',
    '30 min - 1 hr', '1 - 2 hrs', '2 - 3 hrs', '3+ hrs']

  private readonly intensityValues: string[] = ['Low', 'Medium', 'High']
  private readonly consequenceValues: string[] = ['Verbal Redirection', 'Physical assist/prompt',
    'Ignored problem behavior', 'Kept on demand',
    'Verbal reprimand', 'Removed from activity',
    'Given a different activity/task', 'Lost Privilege', 'Sent to room',
    'Given a time out', 'Left alone']

  private readonly refHandlers = {
    peopleRef: (ref) => (this.people = ref),
    behaviorsRef: (ref) => (this.behaviors = ref),
    consequencesRef: (ref) => (this.consequences = ref)
  }

  constructor () {
    super({})
    this.state = this.initialState()
    this.close = this.close.bind(this)
    this.getTime = this.getTime.bind(this)
    this.antecedentRadios = this.antecedentRadios.bind(this)
    this.antecedentOtherText = this.antecedentOtherText.bind(this)
    this.locationRadios = this.locationRadios.bind(this)
    this.durationRadios = this.durationRadios.bind(this)
    this.intensityRadios = this.intensityRadios.bind(this)
    this.save = this.save.bind(this)
    this.postToServer = this.postToServer.bind(this)
    this.validSave = this.validSave.bind(this)
    this.reset = this.reset.bind(this)
  }

  initialState (): ABCState {
    return {
      antecedentOtherDisabled: true,
      messages: [],
      showModal: false,
      user: {
        when: '',
        antecedent: '',
        antecedentOther: '',
        location: '',
        people: [],
        peopleOther: '',
        behavior: [],
        behaviorOther: '',
        duration: '',
        intensity: '',
        consequence: [],
        consequenceOther: ''
      }
    }
  }

  close (): void {
    this.setState({ showModal: false })
  }

  getTime (): void {
    const user = this.state.user
    const time = new Date()
    user.when = time.toLocaleString()
    this.setState({ user })
  }

  antecedentRadios (event): void {
    const antecedent = event.target.value
    const user = this.state.user
    user.antecedent = antecedent
    let antecedentOtherDisabled = this.state.antecedentOtherDisabled
    if (antecedent === 'Other') {
      antecedentOtherDisabled = false
    } else {
      antecedentOtherDisabled = true
      user.antecedentOther = ''
    }
    this.setState({
      user,
      antecedentOtherDisabled
    })
  }

  antecedentOtherText (event): void {
    const antecedentOtherText = event.target.value
    const user = this.state.user
    user.antecedentOther = antecedentOtherText
    this.setState({ user })
  }

  locationRadios (event): void {
    const location = event.target.value
    const user = this.state.user
    user.location = location
    this.setState({ user })
  }

  durationRadios (event): void {
    const duration = event.target.value
    const user = this.state.user
    user.duration = duration
    this.setState({ user })
  }

  intensityRadios (event): void {
    const intensity = event.target.value
    const user = this.state.user
    user.intensity = intensity
    this.setState({ user })
  }

  save (): void {
    const user = this.state.user
    const peoplePull = this.people.pullCurrentState()
    user.people = peoplePull.selected
    user.peopleOther = peoplePull.otherLabelText
    const behaviorPull = this.behaviors.pullCurrentState()
    user.behavior = behaviorPull.selected
    user.behaviorOther = behaviorPull.otherLabelText
    const consequencePull = this.consequences.pullCurrentState()
    user.consequence = consequencePull.selected
    user.consequenceOther = consequencePull.otherLabelText
    this.setState({ user, showModal: true })
    if (this.validSave()) {
      this.postToServer()
    }
  }

  postToServer (): void {
    const data: string = JSON.stringify(this.state.user)
    console.log('ABC data sent ', data)
    const xhr: XMLHttpRequest = new XMLHttpRequest()
    xhr.open('POST', '/saveABC')
    xhr.setRequestHeader('Content-Type', 'application/json')
    xhr.onload = () => {
      const messages: string[] = []
      if (xhr.status >= 200 && xhr.status < 400) {
        console.log('succ ', xhr.responseText)
        messages.push(xhr.responseText)
      } else {
        console.log('unsucc ', xhr.responseText)
        messages.push(xhr.status + ' ' + xhr.statusText)
      }
      this.setState({ messages })
    }
    xhr.onerror = () => {
      console.log(xhr)
    }
    xhr.send(data)
  }

  validSave (): boolean {
    const messages: string[] = []
    if (this.state.user.people.length === 0) {
      messages.push('At least one Person is required to save.')
    } else if (this.state.user.people.indexOf('Other') === 0) {
      if (this.state.user.peopleOther === '') {
        messages.push(
          'For People - Other, the text description of Other must be entered.')
      }
    }
    if (this.state.user.behavior.length === 0) {
      messages.push('At least one Behavior is required to save.')
    } else if (this.state.user.behavior.indexOf('Other') === 0) {
      if (this.state.user.behaviorOther === '') {
        messages.push(
          'For Behavior - Other, the text description of Other must be entered.')
      }
    }
    if (this.state.user.consequence.length === 0) {
      messages.push('At least one Consequence is required to save.')
    } else if (this.state.user.consequence.indexOf('Other') === 0) {
      if (this.state.user.consequenceOther === '') {
        messages.push(
          'For Consequence - Other, the text description of Other must be entered.')
      }
    }
    if (this.state.user.antecedent.length === 0) {
      messages.push('An Antecedent is required to save.')
    } else if (this.state.user.antecedent === 'Other') {
      if (this.state.user.antecedentOther === '') {
        messages.push(
          'For Antecedent - Other, the text description of Other must be entered.')
      }
    }
    if (this.state.user.location.length === 0) {
      messages.push('A Location is required to save.')
    }
    if (this.state.user.duration.length === 0) {
      messages.push('A Duration is required to save.')
    }
    if (this.state.user.intensity.length === 0) {
      messages.push('An Intensity is required to save.')
    }
    if (this.state.user.when.length === 0) {
      messages.push('The date and time of the ABC is required to save.')
    }
    this.setState({ messages })
    return messages.length === 0
  }

  reset (): void {
    this.setState(this.initialState())
    this.people.reset()
    this.behaviors.reset()
    this.consequences.reset()
  }

  render (): ReactNode {
    return (
            <div>
                <h2 className="text-primary">ABC Checklist</h2>
                <div id="abcChecklist">
                    <div className="row">
                        <div className="col-md-1">
                            <h4>When?</h4>
                            <hr/>
                            <input className="btn btn-primary" onClick={this.getTime}
                                   type="button" value="Now" id="now"/>
                            <div id="dateTime">{this.state.user.when}</div>
                        </div>
                        <div className="col-md-4">
                            <h4>Antecedent</h4>
                            <hr/>
                            <b>What happened before the behavior?</b>
                            <br/>
                            {
                                this.antecedentValues.map((antecedent, i) => {
                                  return <div className="radio" key={i}>
                                        <label>
                                            <input type="radio" value={antecedent}
                                                   onClick={this.antecedentRadios}
                                                   checked={this.state.user.antecedent === antecedent}/>{antecedent}
                                        </label>
                                    </div>
                                })
                            }
                            <div className="radio">
                                <label>
                                    <input type="radio"
                                           checked={this.state.user.antecedent === 'Other'}
                                           value="Other"
                                           onClick={this.antecedentRadios}/> Other
                                </label>
                            </div>
                            <input type="text" className="form-control" id="antecedentOther"
                                   value={this.state.user.antecedentOther}
                                   placeholder="Enter Description"
                                   onChange={this.antecedentOtherText}
                                   disabled={this.state.antecedentOtherDisabled}/>
                            <br/> <b>Location</b>
                            <br/>
                            {
                                this.locationValues.map((location, i) => {
                                  return <div className="radio-inline"
                                                key={i}>
                                        <label>
                                            <input type="radio" onClick={this.locationRadios}
                                                   checked={this.state.user.location === location}
                                                   value={location}/>{location}
                                        </label>
                                    </div>
                                })
                            }
                            <br/> <b>People Present</b>
                            <br/>
                            <CheckboxSeries labels={this.peopleValues} selected={this.state.user.people}
                                            ref={this.refHandlers.peopleRef}
                                            otherLabelPlaceholder="Enter Another Person Present"
                                            otherLabelText={this.state.user.peopleOther}/>
                        </div>
                        <div className="col-md-3">
                            <h4>Behavior</h4>
                            <hr/>
                            <b>Select all that apply</b>
                            <CheckboxSeries labels={this.behaviorValues}
                                            ref={this.refHandlers.behaviorsRef}
                                            selected={this.state.user.behavior}
                                            otherLabelPlaceholder="Enter Description"
                                            otherLabelText={this.state.user.behaviorOther}/>
                            <br/> <b>Duration</b>
                            <br/>
                            {
                                this.durationValues.map((duration, i) => {
                                  return <div className="radio-inline" key={i}>
                                        <label>
                                            <input type="radio" onClick={this.durationRadios}
                                                   checked={this.state.user.duration === duration}
                                                   value={duration}/>{duration}
                                        </label>
                                    </div>
                                })
                            }
                            <br/> <b>Intensity</b>
                            <br/>
                            {
                                this.intensityValues.map((intensity, i) => {
                                  return <div className="radio-inline" key={i}>
                                        <label>
                                            <input type="radio" onClick={this.intensityRadios}
                                                   checked={this.state.user.intensity === intensity}
                                                   value={intensity}/>{intensity}
                                        </label>
                                    </div>
                                })
                            }
                        </div>
                        <div className="col-md-4">
                            <h4>Consequence</h4>
                            <hr/>
                            <b>What happened after?</b>
                            <br/>
                            <CheckboxSeries labels={this.consequenceValues}
                                            selected={this.state.user.consequence}
                                            ref={this.refHandlers.consequencesRef}
                                            otherLabelPlaceholder="Enter Description"
                                            otherLabelText={this.state.user.consequenceOther}/>
                        </div>
                    </div>
                    <br/>
                    <div className="form-inline">
                        <div className="form-group">
                            <input className="btn btn-success form-control" type="submit"
                                   onClick={this.save} value="Save" id="save"/>
                        </div>
                        <div className="form-group">
                            <input className="btn btn-danger form-control" type="reset"
                                   onClick={this.reset}
                                   value="Reset" id="reset"/>
                        </div>
                    </div>
                </div>
                <Modal show={this.state.showModal} onHide={this.close}>
                    <Modal.Header>
                        <p>ABC Save Results</p>
                    </Modal.Header>
                    <Modal.Body>
                        <ul>
                            {
                                this.state.messages.map((message, i) => {
                                  return <li key={i}>{message}</li>
                                })
                            }
                        </ul>
                    </Modal.Body>
                    <Modal.Footer>
                        <p style={{ right: 'auto' }}>Click anywhere to continue</p>
                    </Modal.Footer>
                </Modal>
                <div id="abcModal" className="modal">
                    <div className="modal-dialog" role="document">
                        <div className="modal-content">
                            <div className="modal-header">
                                <p>ABC Save Results</p>
                            </div>
                            <div className="modal-body">
                                <ul>
                                    {
                                        this.state.messages.map((message, i) => {
                                          return <li key={i}>{message}</li>
                                        })
                                    }
                                </ul>
                            </div>
                            <div className="modal-footer">
                                <p style={{ right: 'auto' }}>Click anywhere to continue</p>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
    )
  }
}
