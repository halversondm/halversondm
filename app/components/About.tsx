/**
 * Created by Daniel on 6/26/2016.
 */
import * as React from 'react'
import {type ReactNode} from 'react'

export class About extends React.Component<unknown, unknown> {
    render(): ReactNode {
        return (
            <div>
                <h2 className="text-primary">About Dan</h2>
                <h4 className="text-success">My wife calls me a &apos;computer geek&apos;. I don&apos;t
                    mind. It&apos;s true.</h4>
                <p>I received my first computer when I was seven years old. It was a
                    Tandy 1000 with dual floppy drives and 128KB of
                    RAM. We eventually upgraded to 640 KB of RAM, the maximum, and a 20 MB
                    hard drive. A modem was a must. In those
                    days, you could code, play games and check out BBS, but not much else.
                    It&apos;s been said before, and it will be said
                    again the
                    internet changed everything for me. AOL was a glorified BBS. The
                    internet was, and is, open to do with what you
                    please. I turned my hobby
                    into a profession. </p>
                <h4 className="text-success">I love the mechanics of the combustion
                    engine.</h4>
                <p>I have my dad to thank for this. My father spent 45 years in the auto
                    parts business. Half of those years, when I was
                    14 years old and under,
                    as a store owner. I spent a good bit of time selling parts and
                    learning about the engine. He had a machine shop in
                    the back of the store and I spent time taking customers engines apart
                    for valve jobs. It was the coolest thing. One
                    of my favorite smells is uncatalyzed exhaust. These days I do what I
                    can on my cars. The difficult projects require
                    Dad&apos;s help and I enjoy that time we spend together.</p>
                <h4 className="text-success">Kids</h4>
                <p>I have two. Other than this paragraph, I won&apos;t push a page about them
                    in the open internet. I love them and enjoy
                    watching them grow.</p>
                <h4 className="text-success">Wife</h4>
                <p>I met my wife in 1996 at Eastern Illnois University. We were freshman
                    and had mutual friends. We&apos;ve been
                    together ever since. She&apos;s great! We compliment each other very well.
                    At least I think so. We each have our
                    strengths and weaknesses. However, I can&apos;t think of a weakness we
                    share. Well, chocolate, we both love
                    chocolate...</p>
                <h4 className="text-success">Golf</h4>
                <p>I try. I&apos;m getting better. My best friend urged me to join on a
                    regular golf trip with other mutual long time friends
                    in 2014. I bought new clubs that fit me and spent countless hours
                    thinking about hitting that little white
                    ball. Lots of range time too. More range time than course time,
                    actually. I have yet to break 100 or get a birdie,
                    but 2016 is my year.</p>
                <h4 className="text-success">Skiing</h4>
                <p>Like my friends that have been golfing forever, I&apos;ve been skiing
                    since I was 6. I love it. I&apos;m aggressive. I ski
                    backwards downhill. It&apos;s Double Blacks for me. There is something
                    about the speed, cold, and changes in terrain that
                    just light you up.</p>
                <h4 className="text-success">Yard Work</h4>
                <p>I hate it.</p>
            </div>
        )
    }
}
