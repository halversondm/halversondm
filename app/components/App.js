/**
 * Created by Daniel on 6/26/2016.
 */
"use strict";

import React from "react";
import { Link } from "react-router";

let App = React.createClass({

  render() {
    return (
      <div>
        <div className="header">
          <div className="navbar navbar-default navbar-fixed-top" role="navigation">
            <div className="container">
              <div className="navbar-header">
                <button type="button" className="navbar-toggle collapsed"
                        data-toggle="collapse"
                        data-target="#js-navbar-collapse">
                  <span className="sr-only">Toggle navigation</span>
                  <span className="icon-bar"></span>
                  <span className="icon-bar"></span>
                  <span className="icon-bar"></span>
                </button>
                <div className="navbar-brand">@halversondm</div>
              </div>
              <div className="collapse navbar-collapse" id="js-navbar-collapse">
                <ul className="nav navbar-nav">
                  <li><Link to="/home">Home</Link></li>
                  <li><Link to="/apps">Apps</Link></li>
                  <li><Link to="/auto">Auto</Link></li>
                  <li><Link to="/blog">Blog</Link></li>
                  <li><Link to="/resume">Resume</Link></li>
                  <li><Link to="/contact">Contact</Link></li>
                  <li><Link to="/about">About</Link></li>
                </ul>
              </div>
            </div>
          </div>
        </div>
        <div className="theme-showcase container">
          {this.props.children}
          <hr />
          <footer>
            <p>&copy; 2016 dmhweb.org</p>
          </footer>
        </div>
      </div>);
  }
});

export default App;