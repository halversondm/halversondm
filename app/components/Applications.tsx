/**
 * Created by Daniel on 6/26/2016.
 */
import * as React from "react";
import {Link} from "react-router-dom";

export class Applications extends React.Component<{}, {}> {
    render() {
        return (
            <div>
                <h2 className="text-primary">Apps</h2>
                <div className="row">
                    <div className="col-sm-6">
                        <h2 className="text-success">Discount Calculator</h2>
                        <p>I originally wrote this as a Sencha Touch app. However, I lost the
                            code and I can't seem to recover it from the 'compiled' code that
                            Sencha helps you to put together. Anyway, I wrote it for Courtney in
                            2013 because she was working at an outlet store that offered this
                            style of discounts.</p>
                        <p>
                            <Link className="btn btn-info" to="/discountCalculator"
                                  role="button">View details &raquo;</Link>
                        </p>
                        <hr/>
                        <h2 className="text-success">Game</h2>
                        <p>Rock, Paper, Scissors, Lizard, Spock. Play against the computer!
                        </p>
                        <p>
                            <Link className="btn btn-info" to="/game" role="button">View
                                details &raquo;</Link>
                        </p>
                        <hr/>
                        <h2 className="text-success">ABC Checklist</h2>
                        <p>Fill out an Antecedent, Behavior, Consequence Checklist</p>
                        <p>
                            <Link className="btn btn-info" to="/abc" role="button">View
                                details &raquo;</Link>
                        </p>
                        <hr/>
                    </div>
                    <div className="col-sm-6">
                        <h2 className="text-success">Stock Quote</h2>
                        <p>Stock Quote Service</p>
                        <p>
                            <Link className="btn btn-info" to="/stockQuote" role="button">View
                                details &raquo;</Link>
                        </p>
                        <hr/>
                        <h2 className="text-success">Base 64 Encoder and Decoder</h2>
                        <p>Encode and Decode data into and out of Base64 format,
                            respectively.</p>
                        <p>
                            <Link className="btn btn-info" to="/base64" role="button">View
                                details &raquo;</Link>
                        </p>
                        <hr/>
                        <h2 className="text-success">Simple Grade book</h2>
                        <p>Figure out the final grade based on two quizes, one midterm and a
                            final exam</p>
                        <p>
                            <Link className="btn btn-info" to="/grades" role="button">View
                                details &raquo;</Link>
                        </p>
                        <hr/>
                        <h2 className="text-success">URL Builder</h2>
                        <p>Build a URL and launch it</p>
                        <p>
                            <Link className="btn btn-info" to="/urlBuilder" role="button">View
                                details &raquo;</Link>
                        </p>
                        <hr/>
                    </div>
                </div>
            </div>
        );
    }
}
