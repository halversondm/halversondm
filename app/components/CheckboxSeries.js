/**
 * Created by Daniel on 6/30/2016.
 */
"use strict";

import React from "react";

const CheckboxSeries = React.createClass({

  propTypes: {
    labels: React.PropTypes.array,
    selected: React.PropTypes.array,
    otherLabelPlaceholder: React.PropTypes.string,
    otherLabelText: React.PropTypes.string
  },

  getDefaultProps() {
    return {
      labels: [],
      selected: [],
      otherLabelPlaceholder: "",
      otherLabelText: ""
    };
  },

  getInitialState() {
    return {
      otherLabelText: this.props.otherLabelText,
      otherLabelDisabled: true,
      selected: this.props.selected
    };
  },

  toggleSelected(event) {
    var label = event.target.value;
    var selected = this.state.selected;
    var index = selected.indexOf(label);
    if (index > -1) {
      selected.splice(index, 1);
    } else {
      selected.push(label);
    }
    if (label === "Other") {
      this.otherLabelText();
    }
    this.setState({selected: selected});
  },
  otherLabelText() {
    var otherLabelDisabled = this.state.otherLabelDisabled;
    if (otherLabelDisabled) {
      otherLabelDisabled = false;
    } else {
      otherLabelDisabled = true;
    }
    this.setState({otherLabelDisabled: otherLabelDisabled});
  },
  otherLabelChange(event) {
    var otherLabelText = event.target.value;
    this.setState({otherLabelText: otherLabelText});
  },
  render() {
    return <div>
      {
        this.props.labels.map((label, i) => {
          return <div className="checkbox" key={i}>
            <label>
              <input id={label} name="selectedLabel[]" type="checkbox"
                     value={label}
                     checked={this.state.selected.indexOf(label) > -1}
                     onClick={this.toggleSelected}/> {label}
            </label>
          </div>;
        })
      }
      <div className="checkbox">
        <label>
          <input id="otherLabel" type="checkbox" name="selectedLabel[]"
                 checked={this.state.selected.indexOf("Other") > -1}
                 value="Other"
                 onClick={this.toggleSelected}/> Other
        </label>
      </div>
      <input type="text" className="form-control" id="labelOther"
             value={this.state.otherLabelText} onChange={this.otherLabelChange}
             placeholder={this.props.otherLabelPlaceholder}
             disabled={this.state.otherLabelDisabled}/>
    </div>;
  }
});

export default CheckboxSeries;
