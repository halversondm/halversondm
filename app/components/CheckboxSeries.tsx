/**
 * Created by Daniel on 6/30/2016.
 */
import * as React from "react";
import { type ReactNode } from "react";

export interface CheckboxSeriesProps {
  labels: string[];
  selected: string[];
  otherLabelPlaceholder: string;
  otherLabelText: string;
  otherLabelDisabled: boolean;
  toggleSelected: (event) => void;
  setOtherLabelChange: (event) => void;
}

export default function CheckboxSeries({
  labels,
  selected,
  otherLabelPlaceholder,
  otherLabelText,
  otherLabelDisabled,
  toggleSelected,
  setOtherLabelChange,
}: CheckboxSeriesProps): ReactNode {
  return (
    <>
      {labels.map((label, i) => {
        return (
          <div className="checkbox" key={i}>
            <label>
              <input
                id={label}
                name="selectedLabel"
                type="checkbox"
                value={label}
                checked={selected.includes(label)}
                onChange={toggleSelected}
              />{" "}
              {label}
            </label>
          </div>
        );
      })}
      <div className="checkbox">
        <label>
          <input
            id="otherLabel"
            type="checkbox"
            name="otherLabelCheckbox"
            checked={selected.includes("Other")}
            value="Other"
            onChange={toggleSelected}
          />{" "}
          Other
        </label>
      </div>
      <input
        type="text"
        className="form-control"
        id="labelOther"
        value={otherLabelText}
        onChange={setOtherLabelChange}
        placeholder={otherLabelPlaceholder}
        disabled={otherLabelDisabled}
      />
    </>
  );
}
