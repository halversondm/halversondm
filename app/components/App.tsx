/**
 * Created by Daniel on 6/26/2016.
 */
import * as React from "react";
import {Container, Nav, Navbar} from "react-bootstrap";
import {BrowserRouter, Route} from "react-router-dom";
import {LinkContainer} from "react-router-bootstrap";
import {ABC} from "./ABC";
import {About} from "./About";
import {Applications} from "./Applications";
import {Auto} from "./Auto";
import {Base64} from "./Base64";
import {Blog} from "./Blog";
import {Charger} from "./Charger";
import {Contact} from "./Contact";
import {Corvette} from "./Corvette";
import {DiscountCalculator} from "./DiscountCalculator";
import {Game} from "./Game";
import {Grades} from "./Grades";
import {GrandPrix} from "./GrandPrix";
import {Home} from "./Home";
import {Honda} from "./Honda";
import {Resume} from "./Resume";
import {StockQuote} from "./StockQuote";
import {UrlBuilder} from "./UrlBuilder";
import {Yamaha} from "./Yamaha";

export class App extends React.Component<{}, {}> {
    render() {
        return (
            <BrowserRouter>
                <div>
                    <Navbar fixed={"top"} bg="primary" expand="lg">
                        <LinkContainer to="/home"><Navbar.Brand>@halversondm</Navbar.Brand></LinkContainer>
                        <Navbar.Toggle aria-controls="basic-navbar-nav"/>
                        <Navbar.Collapse id="basic-navbar-nav">
                            <Nav>
                                <LinkContainer to="/home"><Nav.Link>Home</Nav.Link></LinkContainer>
                                <LinkContainer to="/apps"><Nav.Link>Apps</Nav.Link></LinkContainer>
                                <LinkContainer to="/auto"><Nav.Link>Auto</Nav.Link></LinkContainer>
                                <LinkContainer to="/blog"><Nav.Link>Blog</Nav.Link></LinkContainer>
                                <LinkContainer to="/resume"><Nav.Link>Resume</Nav.Link></LinkContainer>
                                <LinkContainer to="/contact"><Nav.Link>Contact</Nav.Link></LinkContainer>
                                <LinkContainer to="/about"><Nav.Link>About</Nav.Link></LinkContainer>
                            </Nav>
                        </Navbar.Collapse>
                    </Navbar>
                    <Container>
                        <div>
                            <Route exact={true} path="/" component={Home}/>
                            <Route path="/home" component={Home}/>
                            <Route path="/about" component={About}/>
                            <Route path="/blog" component={Blog}/>
                            <Route path="/resume" component={Resume}/>
                            <Route path="/contact" component={Contact}/>
                            <Route path="/auto" component={Auto}/>
                            <Route path="/honda" component={Honda}/>
                            <Route path="/yamaha" component={Yamaha}/>
                            <Route path="/charger" component={Charger}/>
                            <Route path="/grandprix" component={GrandPrix}/>
                            <Route path="/apps" component={Applications}/>
                            <Route path="/abc" component={ABC}/>
                            <Route path="/base64" component={Base64}/>
                            <Route path="/discountCalculator" component={DiscountCalculator}/>
                            <Route path="/grades" component={Grades}/>
                            <Route path="/game" component={Game}/>
                            <Route path="/stockQuote" component={StockQuote}/>
                            <Route path="/urlBuilder" component={UrlBuilder}/>
                            <Route path="/corvette" component={Corvette}/>
                        </div>
                        <hr/>
                        <footer>
                            <p>&copy; 2021 halversondm.com</p>
                        </footer>
                    </Container>
                </div>
            </BrowserRouter>
        );
    }
}
