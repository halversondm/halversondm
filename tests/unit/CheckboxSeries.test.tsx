import * as React from "react";
import * as renderer from "react-test-renderer";
import {CheckboxSeries} from "../../app/components/CheckboxSeries";

test("Checkbox Series to match", () => {
    const component = renderer.create(<CheckboxSeries labels={["one", "two"]} selected={["one"]}
                                                      otherLabelPlaceholder="test" otherLabelText="anotherTest"/>);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});