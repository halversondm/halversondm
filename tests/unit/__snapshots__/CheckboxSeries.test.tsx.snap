// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Checkbox Series to match 1`] = `"<div><div class="checkbox"><label><input id="one" name="selectedLabel" type="checkbox" value="one"> one</label></div><div class="checkbox"><label><input id="two" name="selectedLabel" type="checkbox" value="two"> two</label></div><div class="checkbox"><label><input id="undefinedOtherLabelCheckbox" type="checkbox" name="undefinedOtherLabelCheckbox" value="Other"> Other</label></div><input type="text" class="form-control" id="undefinedOtherLabelText" placeholder="test" value="anotherTest"></div>"`;
